gc/base/GCExtensionsBase.hpp

```diff
@@ -259,6 +259,7 @@ class MM_GCExtensionsBase : public MM_BaseVirtual {
 	bool largePageFailedToSatisfy;
 #if defined(OMR_GC_DOUBLE_MAP_ARRAYLETS)
 	bool isArrayletDoubleMapRequested;
+	bool isDoubleMapAPIAvailable;
```

Comment:
Changing isDoubleMapAPIAvailable is fine but if we update isArrayletDoubleMapRequested then things might break in OpenJ9. There's an OpenJ9 PR coming after this so I can update it there. How can we do this, meaning, if we rename here OpenJ9 breaks since it won't be able to find the flag. Or we can update all instances in OMR side and just leave the declaration, and once everything is merged then we can delete it? Not sure if this would be the best way, what you think?

Reply:
Renaming is very annoying, and we should try really hard to avoid breaking the builds.  Do you think that renaming `isArrayletDoubleMapRequested` to  `isArrayletDoubleMapEnabled` a good change? It might not be worth it.

If we want to go ahead with renaming, we might be able to do this in 3 commits.
1. OMR: Duplicate `isArrayletDoubleMapRequested` as `isArrayletDoubleMapEnabled`, or (`||`) the fields together
2. J9: switch to use `isArrayletDoubleMapEnabled`
3. OMR: Delete `isArrayletDoubleMapRequested` versions

Reply:
Yes, I agree with the renaming. Better to deal with this now rather than having inconsistent naming or have to deal with it later. I'll update it

Reply:
Having a second look at it, I think for the first variable `isArrayletDoubleMapRequested` might make more sense than `isArrayletDoubleMapEnabled`. This variable is representing if double mapping was requested or not. The fact the double map is enabled or not will depend on this variable and other factors, such as `isDoubleMapAPIAvailable`. With that being said, it does make sense to change the second one to `isArrayletDoubleMapAvailable`. The first variable is also used in `gc/verbose/VerboseHandlerOutput.cpp` to report if double map was requested or not. And if double mapping is enabled or not is dictated by `extensions->indexableObjectModel.isDoubleMappingEnabled()`, instead, does that make sense? What you think?

Reply:
It sounds like `requested` makes more sense here :) thanks for the explanation. Maybe down the line we will have something like:
```c++
bool isArrayletDoubleMapEnabled() { return requested && available; }
```

Reply:
Yes, we actually have that check in OpenJ9 side, `requested && available` to decide if we should actually double map or not. Should I introduce `isArrayletDoubleMapEnabled()` or just later?

Reply:
I think it makes sense to add it to GCExtensions.  Whether you do it now or later is up to you :)

Reply:
Sounds good!